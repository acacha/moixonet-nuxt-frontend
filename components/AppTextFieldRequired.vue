<template>
  <v-text-field
    :value="value"
    :error-messages="errors"
    @input="$v.value.$touch()"
    @blur="$v.value.$touch()"
    :name="name"
    :label="label"
    data-test="text_field_required"
    required
  />
</template>

<script>
import { required } from 'vuelidate/lib/validators'

export default {
  name: 'AppTextFieldRequired',
  validations: {
    name: { required }
  },
  props: {
    label: {
      type: String,
      required: true
    },
    name: {
      type: String,
      required: true
    }
  },
  computed: {
    errors () {
      const errors = []
      if (!this.$v.value.$dirty) { return errors }
      !this.$v.name.required && errors.push('Camp ' + this.name + ' obligatori')
      return errors
    }
  },
  methods: {
    onInput () {
      this.$v.name.$touch()
      this.$emit('input', this.value)
    }
  }
}
</script>
