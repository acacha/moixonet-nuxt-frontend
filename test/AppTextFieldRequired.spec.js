import { mount } from '@vue/test-utils'
import AppTextFieldRequired from '../components/AppTextFieldRequired.vue'

describe('AppTextFieldRequired', () => {
  let opts
  const build = (options) => {
    options = options || opts
    const wrapper = mount(AppTextFieldRequired, options)
    return {
      wrapper
    }
  }
  test('is a Vue instance', () => {
    const { wrapper } = build()
    expect(wrapper.isVueInstance()).toBeTruthy()
  })
  test('renders the component', () => {
    const { wrapper } = build()
    expect(wrapper.element).toMatchSnapshot()
  })
})
