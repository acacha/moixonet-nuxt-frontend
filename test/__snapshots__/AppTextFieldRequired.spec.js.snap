// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppTextFieldRequired renders the component 1`] = `
<h1>
  sada
</h1>
`;
